### ISSUE TITLE
Multiple assemblies/assembly reference?

### ISSUE THREAD

**Issue by fdcastel** [fsprojects/Fable#88]
I'm trying to port my tests from F# to Js the same way Fable does.

Problem is: unlike `Fable.Tests`, my test project have a reference to another project which is also compiled with Fable:

```
MyProj.Core.fsproj

MyProj.Tests.fsproj
  + References MyProj.Core
```

The `.Core` project compiles successfully. However, the `.Tests` projects fails to compile when Fable encounters a type declared into `.Core` project:

`Cannot find replacement for MyType at ...`

where `MyType` is defined in `MyProj.Core`.

There's any plans to make this happen in the future?

As a workaround I can add all units from `.Core` project into `.Tests`. Cheap. But it will work. :)

This will be needed for larger projects, however.

**Comment by alfonsogarciacaro** [comment]
True! We need a way to tell the compiler that a project reference should be compiled as JS imports. Probably this should be done in the `fableconfig.json` file. Right now I'm modifying a bit the way imports/exports work and it's very important to take this into account, thanks a lot for the reminder! :)

**Comment by alfonsogarciacaro** [comment]
I think we have it! I've been working in a refactoring of the imports & exports in the `imports` branch. Could you please have a look and tell me if it fits your needs before merging with `master`? It only works with project references. You must pass a `--refs` argument to link referenced projects with the JS import path that must be used, using the following format: `[Project name without extension]=[JS import path]`

Example:
```
fable src/lib/MyLib.fsproj --outDir out/lib
fable src/another/MyNs.AnotherProject.fsproj --outDir out/another
fable src/main/MyProject.fsproj --outDir out/main --refs MyLib=../lib MyNs.AnotherProject=../another
```

> In `fableconfig.json` you can use a dictionary instead:
```json
{
"refs": {
    "MyLib": "../lib",
    "MyNs.AnotherProject": "../another"
  }
}
```

You can also check [how the tests are compiled now](https://github.com/fsprojects/Fable/blob/imports/build.fsx#L136-L147).

Looking forward to your valuable feedback!

**Comment by fdcastel** [comment]
Great news!! 

Unfortunately I'm having some errors with the /imports branch. 

1) The "console" sample::

Running from /master:

```
Start compilation...
Compiled util.fs at 05:47:26
Compiled index.fs at 05:47:26
node out/index World
Hello dlroW!
```

Running from /imports:

```
Start compilation...
Compiled util.fs at 05:45:51
Compiled index.fs at 05:45:51
node out/index World
ERROR: module.js:341
    throw err;
    ^

Error: Cannot find module 'fable-core'
```


2) Also, projects created with Visual Studio template are giving this error:

```
Start compilation...
The given key was not present in the dictionary. (D:\Temp\FablePluginsTests\NUnit\AssemblyInfo.fs)
```

(I'm trying to make some smaller reproducible sample case -- but maybe this is somewhat related to your findings in #92 ?)



3) This line

https://github.com/fsprojects/Fable/blob/imports/src/tests/package.json#L24

shouldn't be 0.1.6 ?

**Comment by fdcastel** [comment]
Attached is a sample made from Visual Studio template (with an `AssemblyInfo.fs`).

It builds and run fine in `/master`. But fails to do in `/imports` with the error:

```
Start compilation...
The given key was not present in the dictionary. (D:\Temp\FablePluginsTests\NUnit\AssemblyInfo.fs)
```

[TstImports.zip](https://github.com/fsprojects/Fable/files/204155/TstImports.zip)

**Comment by fdcastel** [comment]
Ahhhhnnn. I got it. I'm studying you last commit from master and I see now the "fable-core" package. Please disregard items 1) and 3) above. I'm checking the 2) now.

**Comment by fdcastel** [comment]
Yeah! 2) Worked, too! After I add this package.json:

```
{
  "private": true,
  "engines": {
    "fable": "^0.1.6"
  },
  "dependencies": {
    "fable-core": "0.0.9"
  }
}
```

The project compiled fine (even with AssemblyInfo, this time).

NOW I can look at the core issue you tried to solve. :) Please wait.

(Before, I just have to merge 092f261d69ec0759f21192f649b08d2d2954fc3d with my branches :sweat_smile: )

**Comment by fdcastel** [comment]
Great work!! Everything is working fine! :tada:

The `--refs` also solved another problem for me. Before it I had to output files in several folders mapping the original F# folder structure. Something like:

```
.\MyProj.Core.fsproj
.\Ext\MyProj.Ext.fsproj        (references ..\MyProj.Core)
.\Tests\MyProj.Tests.fsproj    (references ..\MyProj.Core)

Compiles to

.\*.js
.\Ext\*.js
.\Tests\*.js
```

Now I'm just throwing all generated .js files in one folder and using  `--refs MyProj.Core.fsproj=.`  :+1:

**Comment by alfonsogarciacaro** [comment]
Super! Sorry, I was busy yesterday and couldn't reply to you properly. I forgot to tell you the samples weren't fixed at the time. This is fixed now together with the other problems you reported :+1:

Yes, I moved the core JS library to a separate NPM package. I wanted to make Fable work without dependencies, but I realised at the end most users are going to use NPM for other dependencies anyway. This should prevent duplication of the core library when working with multiple projects.
